writer.go

```diff
@@ -141,31 +138,56 @@ func (w *writeWorker) done(reqs []*request, err error) {
 	}
 }
 
-func (w *writeWorker) writeToLSM(b *request) error {
-	if len(b.Ptrs) != len(b.Entries) {
-		return errors.Errorf("Ptrs and Entries don't match: %+v", b)
+func (w *writeWorker) writeToLSM(entries []*Entry, ptrs []valuePointer) error {
+	if len(ptrs) != len(entries) {
+		return errors.Errorf("Ptrs and Entries don't match: %v, %v", entries, ptrs)
+	}
+
+	var i uint64
+	var err error
+	for err = w.ensureRoomForWrite(); err == errNoRoom; err = w.ensureRoomForWrite() {
```

Comment:
Is it possible that the skipList size + pendingList size is greater than max table size, but we didn't count pendingList size here, then later when we merge pending list to skipList, it exceeds the max table size?

Reply:
`MemTable.MemSize()` will count size of `pendingList`. It use `MaxNodeSize + nodeAlign + len(key) + len(value)` as the estimated size of an entry.